qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -432,6 +474,55 @@ def parse_offset_get(s: str) -> Tuple[float, int]:
                            get_cmd='SPTS ?',
                            get_parser=int)
 
+        self.add_parameter('sweep_start',
+                           unit='',
+                           initial_value=0,
+                           get_cmd=None,
+                           set_cmd=None)
+
+        self.add_parameter('sweep_stop',
+                           unit='',
+                           initial_value=1,
+                           get_cmd=None,
+                           set_cmd=None)
+
+        self.add_parameter('sweep_n_points',
+                           unit='',
+                           initial_value=10,
+                           vals=Numbers(1, 16e3),
+                           get_cmd=None,
+                           set_cmd=None)
+
+        self.add_parameter('sweep_setpoints',
+                           parameter_class=GeneratedSetPoints,
+                           startparam=self.sweep_start,
+                           stopparam=self.sweep_stop,
+                           numpointsparam=self.sweep_n_points,
+                           vals=Arrays(shape=(self.buffer_npts.get,)))
+
+        # Channel setup
+        for ch in range(1, 3):
+
+            # detailed validation and mapping performed in set/get functions
+            self.add_parameter(f'ch{ch}_ratio',
+                               label=f'Channel {ch} ratio',
+                               get_cmd=partial(self._get_ch_ratio, ch),
+                               set_cmd=partial(self._set_ch_ratio, ch),
+                               vals=Strings())
+            self.add_parameter(f'ch{ch}_display',
+                               label=f'Channel {ch} display',
+                               get_cmd=partial(self._get_ch_display, ch),
+                               set_cmd=partial(self._set_ch_display, ch),
+                               vals=Strings())
+            self.add_parameter(f'ch{ch}_databuffer',
+                               channel=ch,
+                               parameter_class=ChannelBuffer)
+            self.add_parameter(f'ch{ch}_datatrace',
+                               channel=ch,
+                               vals=Arrays(shape=(self.buffer_npts.get,)),
+                               setpoints=(self.sweep_setpoints,),
+                               parameter_class=ChannelTrace)
```

Comment:
Perhaps we should consider if it makes sense to replace this with a InstrumentChannel for each channel. That will be a breaking change but it would bring the instrument more in line with Other instruments. @astafan8 what do you think

Reply:
yes, i had the same feeling when reading the PR. we could leave this driver as is, and make another one next to it in a file called SR830_channels.py, similar to some other drivers.

Reply:
Should I start making a version using Channels?
The use of channels is only relevant if you write to the buffer, so I would prefer to call it SR830_buffered.py 
Are you sure we want to have two drivers for the same instrument?
It will create some confusion for the user of which one to use. 
Could we add it as a non-breaking change, but with duplicated functionality?  
If you want we could have a short call discussing the best strategy?

Reply:
Hi @jenshnielsen  and @astafan8 have you concluded anything regarding if I should change the current driver to include channels or I should make a new one?

Reply:
I think that is fine for now. It would be relative little effort to copy and convert this to a channel version